cmd/main.go

```diff
@@ -12,18 +14,66 @@ func main() {
 	flag.Parse()
 	args := flag.Args()
 	if len(args) < 2 {
-		fmt.Println("please provide IP address and port")
+		fmt.Printf("please provide IP address and port\n")
 		os.Exit(1)
 	}
-	ipAddr := args[0]
-	port := args[1]
-	fmt.Printf("ip %s port %s\n", ipAddr, port)
 
-	_, err := client.NewTcpClient(ipAddr, port)
+	// We might to use localhost...
+	// ip := net.ParseIP(args[0])
+	// if ip == nil {
+	// 	fmt.Printf("invalid IP address: %s\n", args[0])
+	// 	os.Exit(1)
+	// }
+
+	_, err := strconv.ParseUint(args[1], 10, 16) // 0 <= port <= 65535
+	if err != nil {
+		fmt.Printf("invalid port %s, should be between 0 and 65535\n", args[1])
+		os.Exit(1)
+	}
+
+	addr := net.JoinHostPort(args[0], args[1])
+	fmt.Printf("connecting to %s\n", addr)
+
+	c, err := client.NewTCPClient(addr)
 	if err != nil {
 		fmt.Println(err)
 		os.Exit(1)
 	}
 
+	// Send name
+	err = c.SendName("langorou")
+	if err != nil {
+		fmt.Printf("error: %s\n", err)
+		os.Exit(1)
+	}
+
+	// Receive SET
+	err = c.ReceiveSpecificCommand(client.SET)
+	if err != nil {
+		fmt.Printf("error: %s\n", err)
+		os.Exit(1)
+	}
+
+	// Receive HUM
+	err = c.ReceiveSpecificCommand(client.HUM)
+	if err != nil {
+		fmt.Printf("error: %s\n", err)
+		os.Exit(1)
+	}
+
+	// Receive HME
+	err = c.ReceiveSpecificCommand(client.HME)
+	if err != nil {
+		fmt.Printf("error: %s\n", err)
+		os.Exit(1)
+	}
+
+	// Receive MAP
+	err = c.ReceiveSpecificCommand(client.MAP)
+	if err != nil {
```

Comment:
nit: It would be a bit more readable like this

```suggestion
	failIf(err, msg)
```